src/App.tsx

```diff
@@ -1,26 +1,208 @@
-/* eslint-disable max-len */
+/* eslint-disable react-hooks/rules-of-hooks */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, getTodos, deleteTodos, createTodos } from './api/todos';
+import { Filter, Todo, NewTodos } from './types/Todo';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { ErrorComponent } from './components/ErrorComponent';
+import { TodoItem } from './components/TodoItem';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [newTodo, setNewTodo] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [selectedIdTodo, setSelectedIdTodo] = useState<number[]>([]);
+
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [todoId, setTodoId] = useState(0);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const [selectedFilter, setSelectedFilter] = useState<Filter>(Filter.ALL);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  // #region functions
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setTimeout(() => setErrorMessage(''), 3000);
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const filteredTodos = useMemo(() => {
+    switch (selectedFilter) {
+      case Filter.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+
+      case Filter.COMPLETED:
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return todos;
+    }
+  }, [selectedFilter, todos]);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const input = e.target.value;
+
+    setNewTodo(input);
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (newTodo.trim() === '') {
+      setErrorMessage('Title should not be empty');
+      setTimeout(() => setErrorMessage(''), 3000);
+    } else {
+      setIsSubmitting(true);
+      setTempTodo({ id: 0, title: newTodo, completed: false, userId: USER_ID });
```

Comment:
Add it to pass failed test
```suggestion
      setTempTodo({ id: 0, title: newTodo.trim(), completed: false, userId: USER_ID });
```